plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,502 @@
+    state: query
+  register: query_uuid
+
+- name: Query all IGMP Interface Policies in the template
+  cisco.mso.ndo_tenant_igmp_interface_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: TestTenantTemplate
+    state: query
+  register: query_all
+
+- name: Delete an IGMP Interface Policy with the name
+  cisco.mso.ndo_tenant_igmp_interface_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: TestTenantTemplate
+    name: test_igmp_interface_policy
+    state: absent
+
+- name: Delete an IGMP Interface Policy with the UUID
+  cisco.mso.ndo_tenant_igmp_interface_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: TestTenantTemplate
+    uuid: "{{ igmp_interface_policy.current.uuid }}"
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate, KVPair
+from ansible_collections.cisco.mso.plugins.module_utils.constants import ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP
+from ansible_collections.cisco.mso.plugins.module_utils.utils import append_update_ops_data
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True, aliases=["tenant_template"]),
+        name=dict(type="str", aliases=["igmp_interface_policy"]),
+        uuid=dict(type="str"),
+        description=dict(type="str"),
+        version3_asm=dict(type="str", aliases=["allow_version3_asm"], choices=["enabled", "disabled"]),
+        fast_leave=dict(type="str", choices=["enabled", "disabled"]),
+        report_link_local_groups=dict(type="str", choices=["enabled", "disabled"]),
+        igmp_version=dict(type="str", choices=["v2", "v3"]),
+        group_timeout=dict(type="int"),
+        query_interval=dict(type="int"),
+        query_response_interval=dict(type="int"),
+        last_member_count=dict(type="int"),
+        last_member_response_time=dict(type="int"),
+        startup_query_count=dict(type="int"),
+        startup_query_interval=dict(type="int"),
+        querier_timeout=dict(type="int"),
+        robustness_variable=dict(type="int"),
+        state_limit_route_map_uuid=dict(type="str", aliases=["state_limit_route_map_for_multicast"]),
+        state_limit_route_map=dict(
+            type="dict",
+            options=dict(
+                name=dict(type="str", required=True),
+                template=dict(type="str", required=True),
+            ),
+            aliases=["state_limit_route_map_policy", "state_limit_route_map_policy_multicast"],
+        ),
+        report_policy_route_map_uuid=dict(type="str", aliases=["report_policy_route_map_for_multicast"]),
+        report_policy_route_map=dict(
```

Comment:
report_policy_route_map=dict(type="str") with a proper alias if necessary

Reply:
This attribute will be kept as dictionary to accommodate  any future changes.